### ISSUE TITLE
Better coordination for Chromium updates

### ISSUE THREAD

**Issue by Eloston** [Eloston/ungoogled-chromium#1043]
There are several developers who provide updates to new Chromium, but we don't have a way of coordinating them.

I think a simple way to solve this is to create issues titled "Updating to Chromium x.x.x.x" requesting for someone to claim it (perhaps pinning the issue too). Additionally, this can serve as a thread to discuss issues or questions while updating.

Ideally, these issues should be auto-generated whenever a new stable version of Chromium releases (as shown on https://omahaproxy.appspot.com, for example). I'd like an automated way of creating these issues, preferably using a free service. Anyone have ideas?

**Comment by jstkdng** [comment]
We could create a scheduled github workflow that scrapes the chromium releases webpages and creates a issue.
In that issue anyone who wants to work on it can say that they are and anyone else who would like to work on it can try to help someone or do his own thing as a case can happen that someone claims and never delivers.

**Comment by Eloston** [comment]
> We could create a scheduled github workflow that scrapes the chromium releases webpages and creates a issue.

Huh, did not know that. This'll work.

> In that issue anyone who wants to work on it can say that they are and anyone else who would like to work on it can try to help someone or do his own thing as a case can happen that someone claims and never delivers.

For now, I'll give the benefit of the doubt and hope whoever claims it is willing and able to see it through. I might also suggest that they create a draft PR so there's better visibility.

**Comment by wchen342** [comment]
I was trying to find the workflow file but wasn't able to. @Zoraver are you creating the issues manualy now?

**Comment by Eloston** [comment]
@wchen342 Yeah the issues are still created manually. There's a template for them.

**Comment by PF4Public** [comment]
> @wchen342 Yeah the issues are still created manually.

Could we revive #1219 and/or improve it to be helpful?

**Comment by wchen342** [comment]
Why was that PR closed?

**Comment by PF4Public** [comment]
My implementation was functional, but had some flaws in several aspects. There were neither alternatives, nor flaw-fixes suggested, so I closed it.

**Comment by networkException** [comment]
I'm not sure how to solve the issue that different platforms might use different versions, I guess sticking with desktop (linux perhaps) should be enough for automatic issues, everything else can be done manually.

Every blogpost about a new stable release seems to include a link to the scm changelog, the link is the same as clicking on "View Change Log" on <https://omahaproxy.appspot.com/> (aka the redirect result of <https://omahaproxy.appspot.com/cl?os=linux&channel=stable>). Maybe a good way to match release blogposts is to look for link tags with an url to the scm log?

**Comment by PF4Public** [comment]
> I guess sticking with desktop (linux perhaps) should be enough for automatic issues, everything else can be done manually.

I think this would be easiest and the most robust solution. So long that https://omahaproxy.appspot.com/linux gives us what we need :)

**Comment by Eloston** [comment]
@PF4Public That sounds good to me!

**Comment by PF4Public** [comment]
@Ahrotahn What is your opinion on this? I'm asking because it is already a second time, that you start preparing a release ahead of time, way earlier, than omaha updates and a scripted issue could be created. Is it necessary to auto-create "Updating to" issues in your opinion?

**Comment by Ahrotahn** [comment]
It would still be useful for the minor releases since they are often pretty sporadic.  I'm not familiar enough with actions to know what's possible, but maybe if there's a way to get the stable cut date from https://chromiumdash.appspot.com/schedule then that would be useful for the major updates.

**Comment by wchen342** [comment]
What's the problem if we just poll the omaha server? It will not be beforehand but that is not a major concern I think, at least not for minor updates.

**Comment by PF4Public** [comment]
> What's the problem if we just poll the omaha server?

A single problem that I foresee is that someone creates "Update to Chromium 96.0.4664.x" and then the script creates "Update to Chromium 96.0.4664.45". Is it a "bid deal"?

**Comment by wchen342** [comment]
I think if we have a script automatically pull versions than there is no need for others to create these issues anymore?

**Comment by PF4Public** [comment]
@wchen342 https://github.com/Eloston/ungoogled-chromium/issues/1043#issuecomment-968906031


> I'm asking because it is already a second time, that you start preparing a release ahead of time, way earlier, than omaha updates and a scripted issue could be created.

**Comment by wchen342** [comment]
@PF4Public But that's because we do not have a script right now. If @Ahrotahn knows we have a script than he may make the PR and link to the issue when it is created. If someone else still created a duplicate issue than that can be closed as duplicate.

**Comment by PF4Public** [comment]
> If @Ahrotahn knows we have a script than he may make the PR and link to the issue when it is created. If someone else still creates a duplicate issue than that can be closed as duplicate.

Fair enough
